### PR TITLE
Adding the BackgroundFetchEvent interface

### PR DESCRIPTION
**Pull request by rachelandrew**
Adds the BackgroundFetchEvent interface.

Reviewer: @jpmedley

### CODE REVIEW

File: files/en-us/web/api/backgroundfetchevent/index.html [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+<h2 id="Constructor">Constructor</h2>
+
+<dl>
+ <dt>{{domxref("BackgroundFetchEvent.BackgroundFetchEvent()", "BackgroundFetchEvent()")}}</dt>
+ <dd>Creates a new <code>BackgroundFetchEvent</code> object. This constructor is not typically used, as the browser creates these objects itself and provides them to background fetch event callbacks.</dd>
+</dl>
+
+<h2 id="Properties">Properties</h2>
+
+<p><em>Inherits properties from its ancestor, {{domxref("Event")}}</em>.</p>
```
Comment (jpmedley) [inline files/en-us/web/api/backgroundfetchevent/index.html:25]:
```suggestion
<p><em>Inherits properties from its ancestor, {{domxref("ExtendableEvent")}}</em>.</p>
```

File: files/en-us/web/api/backgroundfetchevent/index.html [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+<p>None.</p>
+
+<h2 id="Methods">Methods</h2>
+
+<p>None.</p>
+
+<h2 id="Examples">Examples</h2>
+
+<p>In this example, if the user clicks on the UI displaying the download progress, a new window will open. The current {{domxref("BackgroundFetchRegistration")}} is returned by calling <code>event.registration</code>.</p>
```
Comment (jpmedley) [inline files/en-us/web/api/backgroundfetchevent/index.html:42]:
```suggestion
<p>In this example, if the user clicks on the user interface displaying the download progress, a new window will open. The current {{domxref("BackgroundFetchRegistration")}} is returned by calling <code>event.registration</code>.</p>
```

File: files/en-us/web/api/backgroundfetchevent/registration/index.html [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+<h2 id="Syntax">Syntax</h2>
+
+<pre class="syntaxbox">let registration = BackgroundFetchEvent.registration;</pre>
+
+<h3>Value</h3>
+<p>A  {{domxref("BackgroundFetchRegistration")}}.</p>
+
+<h2 id="Examples">Examples</h2>
+
+<p>In this example, if the user clicks on the UI displaying the download progress, this fires the <code>onbackgroundfetchclick</code> event. The current {{domxref("BackgroundFetchRegistration")}} is returned by calling <code>event.registration</code>.</p>
```
Comment (jpmedley) [inline files/en-us/web/api/backgroundfetchevent/registration/index.html:24]:
```suggestion
<p>In this example, if the user clicks on the user interface displaying the download progress, this fires the <code>onbackgroundfetchclick</code> event. The current {{domxref("BackgroundFetchRegistration")}} is returned by calling <code>event.registration</code>.</p>
```

**Review by jpmedley — CHANGES_REQUESTED** [review]

File: files/en-us/web/api/backgroundfetchevent/index.html [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+  - Interface
+  - Reference
+  - BackgroundFetchEvent
+---
+<div>{{DefaultAPISidebar("Background Fetch API")}}</div>
+
+<p class="summary">The <strong><code>BackgroundFetchEvent</code></strong> interface of the {{domxref('Background Fetch API','','',' ')}} is the event type for background fetch events dispatched on the {{domxref("ServiceWorkerGlobalScope", "service worker global scope")}}.</p>
+
+<p>It is the event type passed to <code>onbackgroundfetchabort</code> and <code>onbackgroundfetchclick</code>.</p>
+
```
[the remark this answers is not in the document]
Comment (jpmedley) [reply files/en-us/web/api/backgroundfetchevent/index.html:15]: It looks like `ExtendableEvent` is more like an abstract base class and isn't used anywhere directly. My suggestion is the standard approach for event inheritance, but I don't think it's correct for this event. Don't merge this suggestion. I've made a note to bring it up in Monday's editorial meeting.

[the remark this answers is not in the document]
Comment (rachelandrew) [reply files/en-us/web/api/backgroundfetchevent/index.html:15]: There are some other interfaces which extend ExtendableEvent but have been documented to extend Event - FetchEvent is an example https://developer.mozilla.org/en-US/docs/Web/API/FetchEvent which is why I followed suit.

[the remark this answers is not in the document]
Comment (jpmedley) [reply files/en-us/web/api/backgroundfetchevent/index.html:15]: I forgot to bring it up. Then we'll stick with extending event. We'll need to leave out the inheritance diagram. That's why I'm still going to bring this up.

**Review by jpmedley — APPROVED** [review]

**Pull request merged**